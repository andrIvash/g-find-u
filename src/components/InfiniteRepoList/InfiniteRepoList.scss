@import '../../styles/abstracts/functions';

$margin: rem(10);

.iRepoList {
    .iRepoList__repoCard {
        background-color: var(--app-bgcard-color);
        margin: $margin 0 ;

        .repoCard__fork, .repoCard__lang, .repoCard__time {
            color: var(--app-info-color);
        }
    }
}