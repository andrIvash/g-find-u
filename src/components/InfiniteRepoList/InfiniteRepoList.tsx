import React from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import { IProps } from './index';
import { LOADER_MESSAGE } from '../../resources/constants';
import RepoCard from '../RepoCard';

import './InfiniteRepoList.scss';

export const InfiniteRepoList = (props: IProps) => {
    const { repos, next, hasMore, targetId } = props;
    return (
        <InfiniteScroll
            className='iRepoList'
            dataLength={repos.length}
            next={next}
            hasMore={hasMore}
            loader={LOADER_MESSAGE}
            scrollableTarget={targetId ? targetId : `scrollableDiv`}
            style={{
                overflow: 'hidden'
            }}
        >
            {repos.map((repo, index) => (
                <RepoCard
                    className='iRepoList__repoCard'
                    key={`${repo.link}${index}`}
                    title={repo.title}
                    repoLink={repo.repoLink}
                    isFork={repo.isFork}
                    stars={repo.stars}
                    desc={repo.desc}
                    logoColor={repo.logoColor}
                    lang={repo.lang}
                    time={repo.time}
                />
            ))}
        </InfiniteScroll>
    );
};

export default InfiniteRepoList;