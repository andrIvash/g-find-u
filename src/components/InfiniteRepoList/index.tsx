import { InfiniteRepoList } from './InfiniteRepoList';

export type Repo = any;

export interface IProps {
    repos: Repo[];
    next: () => void;
    hasMore: boolean;
    targetId?: string;
}


export default InfiniteRepoList;