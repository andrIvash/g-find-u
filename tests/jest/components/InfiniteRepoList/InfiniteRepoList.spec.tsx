import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import InfiniteRepoList from '../../../../src/components/InfiniteRepoList';

jest.mock('../../../../src/components/RepoCard', () => ({ title }) => (
    <div data-testid='mock-repo-card'>{title}</div>
));

describe('InfiniteRepoList Component', () => {
    const mockRepos = [
        { title: 'Repo 1' },
        { title: 'Repo 2' },
    ];

    it('renders without crashing', () => {
        render(<InfiniteRepoList repos={mockRepos} next={jest.fn()} hasMore={true} />);
        expect(screen.getByText('Repo 1')).toBeInTheDocument();
    });

    it('renders the correct number of RepoCard components', () => {
        render(<InfiniteRepoList repos={mockRepos} next={jest.fn()} hasMore={true} />);
        const cards = screen.getAllByTestId('mock-repo-card');
        expect(cards.length).toBe(mockRepos.length);
    });

    it('passes correct props to RepoCard components', () => {
        render(<InfiniteRepoList repos={mockRepos} next={jest.fn()} hasMore={true} />);
        expect(screen.getByText('Repo 1')).toBeInTheDocument();
        expect(screen.getByText('Repo 2')).toBeInTheDocument();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<InfiniteRepoList repos={mockRepos} next={jest.fn()} hasMore={true} />);
        expect(asFragment()).toMatchSnapshot();
    });
});