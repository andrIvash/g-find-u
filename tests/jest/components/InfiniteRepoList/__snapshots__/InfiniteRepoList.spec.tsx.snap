// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfiniteRepoList Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component iRepoList"
      style="height: auto; overflow: hidden;"
    >
      <div
        data-testid="mock-repo-card"
      >
        Repo 1
      </div>
      <div
        data-testid="mock-repo-card"
      >
        Repo 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
